src/frontend/screens/Sync/ProjectSyncDisplay.tsx

```diff
@@ -0,0 +1,270 @@
+import {Bar as ProgressBar} from 'react-native-progress';
+import {useProject} from '../../hooks/server/projects';
+import {SyncState, useSyncProgress} from '../../hooks/useSyncState';
+import ObservationsProjectImage from '../../images/ObservationsProject.svg';
+import {
+  BLACK,
+  COMAPEO_BLUE,
+  DARK_GREY,
+  LIGHT_GREY,
+  MEDIUM_GREY,
+  WHITE,
+} from '../../lib/styles';
+import {ScreenContentWithDock} from '../../sharedComponents/ScreenContentWithDock';
+import {Button} from '../../sharedComponents/Button';
+import {Text} from '../../sharedComponents/Text';
+import {StopIcon, SyncIcon, WifiIcon} from '../../sharedComponents/icons';
+import {useQueryClient} from '@tanstack/react-query';
+import {OBSERVATION_KEY} from '../../hooks/server/observations';
+
+const m = defineMessages({
+  deviceName: {
+    id: 'screens.Sync.ProjectSyncDisplay.deviceName',
+    defaultMessage: 'Your device name is {name}',
+  },
+  devicesNearby: {
+    id: 'screens.Sync.ProjectSyncDisplay.devicesNearby',
+    defaultMessage:
+      '{count} {count, plural, one {device} other {devices}} nearby/connected',
+  },
+  buttonTextSync: {
+    id: 'screens.Sync.ProjectSyncDisplay.buttonTextSync',
+    defaultMessage: 'Sync',
+  },
+  buttonTextStop: {
+    id: 'screens.Sync.ProjectSyncDisplay.buttonTextStop',
+    defaultMessage: 'Stop',
+  },
+  buttonTextDone: {
+    id: 'screens.Sync.ProjectSyncDisplay.buttonTextDone',
+    defaultMessage: "You're all caught up",
+  },
+  noDevicesSyncing: {
+    id: 'screens.Sync.ProjectSyncDisplay.noDevicesSyncing',
+    defaultMessage: 'No Devices are Syncing',
+  },
+  devicesWaitingToSync: {
+    id: 'screens.Sync.ProjectSyncDisplay.devicesWaitingToSync',
+    defaultMessage:
+      '{count} {count, plural, one {Device} other {Devices}} Waiting to Sync with you',
+  },
+  devicesSyncing: {
+    id: 'screens.Sync.ProjectSyncDisplay.devicesSyncing',
+    defaultMessage:
+      'Syncing with {count} {count, plural, one {Device} other {Devices}}',
+  },
+  upToDate: {
+    id: 'screens.Sync.ProjectSyncDisplay.upToDate',
+    defaultMessage: 'Up to Date!\nNo data to Sync',
+  },
+  syncing: {
+    id: 'screens.Sync.ProjectSyncDisplay.syncing',
+    defaultMessage: 'Syncing…',
+  },
+  syncProgress: {
+    id: 'screens.Sync.ProjectSyncDisplay.syncProgress',
+    defaultMessage: '{value}%',
+  },
+});
+
+export const ProjectSyncDisplay = ({
+  syncState,
+  projectName,
+  deviceName,
+}: {
+  syncState: SyncState;
+  projectName: string;
+  deviceName: string;
+}) => {
+  const {formatMessage: t} = useIntl();
+
+  const project = useProject();
+  const queryClient = useQueryClient();
+
+  const {connectedPeers, data, initial} = syncState;
+  const isSyncDone = !initial.dataToSync && !data.dataToSync;
+
+  React.useEffect(() => {
+    if (isSyncDone) {
+      project.$sync.stop();
+      queryClient.invalidateQueries({queryKey: [OBSERVATION_KEY]});
+    }
+  }, [isSyncDone, project, queryClient]);
```

Comment:
discussed elsewhere but this check should be done when leaving the screen (e.g. `navigation.addListener('beforeRemove', ...)`